ratis-common/src/main/java/org/apache/ratis/util/FileUtils.java

```diff
@@ -77,7 +77,7 @@ static void move(File src, File dst) throws IOException {
 
   static void move(Path src, Path dst) throws IOException {
     LogUtils.runAndLog(LOG,
-        () -> Files.move(src, dst),
+        () -> Files.move(src, dst, StandardCopyOption.ATOMIC_MOVE),
```

Comment:
According to [javadoc](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#move-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-), this would fail with exception if move cannot be performed as an atomic operation.  If I understand correctly, moving across filesystems is such a non-atomic operation, performed as "copy and delete" instead.

Should we retry without the option in case of `AtomicMoveNotSupportedException`?

Reply:
Thanks for your review! Indeed we may encounter cross disk move, so I'll add the fallback retry logic.

Reply:
Thanks for updating the patch, LGTM.